posts/2024/2024_07_12_Inigo_week_7.rst

```diff
@@ -0,0 +1,37 @@
+Week 7 into GSoC 2024: Starting to see the light at the end of the VAE
+======================================================================
+
+
+.. post:: July 12 2024
+    :author: Iñigo Tellaetxe
+    :tags: google
+    :category: gsoc
+
+
+
+What I did this week
+~~~~~~~~~~~~~~~~~~~~
+
+Finally, I figured out how to solve the ``nan`` value problem in the VAE training. As I suspected, the values that the ``ReparametrizationTrickSampling`` layer was getting were too big for the exponential operations. I use the exponential operation because I am treating the Encoder output as the log variance of the latent space distribution, and for sampling we need the standard deviation. We use the log variance instead of the standard deviation for avoiding computing logarithms.
+The solution consisted of two measures: first, clipping the values of the exponential operations to the ``[-1e10, 1e10]`` range. Then, adding batch normalization layers after each convolution in the Encoder enforcing mean=0 and variance=1, avoids the values from getting too big. I did not use batch normalization in the fully connected layers that output the mean and the log variance of the latent distribution because I did not want to constrain the latent space too much with distributions with the mentioned characteristics. These layers should capture other characteristics of the data distribution.
```

Comment:
`(...) range. Then, adding batch normalization layers after each convolution in the Encoder enforcing mean=0 and variance=1, avoids the values from getting too big.`

`(...) range; second, adding batch normalization layers after each convolution in the Encoder enforces a 0 mean and unit variance, preventing large values at the output.`